ironfish/src/consensus/verifier.ts

```diff
@@ -519,6 +531,26 @@ export class Verifier {
 
     return { valid: true }
   }
+
+  /**
+   * Given a transaction, verify that the hash is not present in the blockchain
+   * already. Most of the time, we can count on spends being present, so regular
+   * double-spend checks are sufficient. However, if the minimum fee is 0,
+   * transactions that do not contain spends could be replayable in some
+   * scenarios.
+   */
+  verifyUnseenTransaction(
+    transaction: Transaction,
+    tx?: IDatabaseTransaction,
+  ): Promise<VerificationResult> {
+    return this.chain.db.withTransaction(tx, async (tx) => {
+      if (await this.chain.transactionHashToBlockHash.has(transaction.hash(), tx)) {
+        return { valid: false, reason: VerificationResultReason.DOUBLE_SPEND }
```

Comment:
nit: I think with a default min fee of 1 currently in our consensus, we'll implicitly always have a spend in a transaction so the double spend reason would hold true. However, all our current double spend error messages arise from checking associated nullifiers. How do you feel about using something like `VerificationResultReason.EXISTING_TRANSACTION`? It feels cleaner to me, but I don't feel that strongly about it.

Reply:
Good call - I'll add a new error type.